Binder/CMake does not support out-of-place builds

Building Bilder following instructions [1] for the case "Installation with pre-installed LLVM".
Works for me _in place_, i.e. `cmake .`. Does not work _out of place_, like
```
mkdir build
cd build
cmake ..
```
as make will fail with 
```
make[2]: Entering directory '/G/ext/binder/build'
cd /G/ext/binder/test && C_INCLUDE_PATH=/usr/lib/llvm-13/lib/clang/13.0.1/include CPLUS_INCLUDE_PATH=/usr/lib/llvm-13/lib/clang/13.0.1/include /G/ext/binder/build/source/binder --bind "" -max-file-size=100000 --root-module T50_namespace_binder --prefix /G/ext/binder/build/test/ --config T50.namespace_binder.config --single-file --annotate-includes T50.namespace_binder.hpp.include -- -x c++ -std=c++11 -I . -I /G/ext/binder/test -isystem / -I /usr/lib/llvm-13/include -iwithsysroot/usr/lib/llvm-13/lib/clang/13.0.1/include
error: no such file or directory: '/G/ext/binder/test/T50.namespace_binder.hpp.include'
error: no input files
error: unable to handle compilation, expected exactly one compiler job in ''
Error while processing /G/ext/binder/test/T50.namespace_binder.hpp.include.
make[2]: *** [test/CMakeFiles/targetT50_namespace_bindercpp.dir/build.make:73: test/CMakeFiles/targetT50_namespace_bindercpp] Error 1
make[2]: Leaving directory '/G/ext/binder/build'
make[1]: *** [CMakeFiles/Makefile2:902: test/CMakeFiles/targetT50_namespace_bindercpp.dir/all] Error 2
make[1]: Leaving directory '/G/ext/binder/build'
make: *** [Makefile:149: all] Error 2
```

In-place compilation clutters the source directory with non-source files, which makes it difficult to revert to initial state or to create pull requests. Out-of-place compilation is simple and clean and should be supported, and also should be recommended in the docs.

[1] https://cppbinder.readthedocs.io/en/latest/install.html

--- a/test/CMakeLists.txt
+++ b/test/CMakeLists.txt
@@ -11,6 +11,7 @@ message( STATUS "binder test: building ${stestnamenodot}.cpp for ${stestname}" )
 if (BINDER_MOCK_TEST)
   file(GENERATE OUTPUT ${CMAKE_CURRENT_BINARY_DIR}/${stestnamenodot}.cpp INPUT ${CMAKE_CURRENT_SOURCE_DIR}/${stestname}.ref )
 else()
+  make_directory(${CMAKE_CURRENT_BINARY_DIR}/test)
   SET(${stestname}_config_flag "")
   if(EXISTS "${CMAKE_CURRENT_SOURCE_DIR}/${stestname}.config")
     SET(${stestname}_config_flag --config ${stestname}.config )
@@ -21,14 +22,14 @@ else()
     #The first two expressions below are for older clang
     COMMAND C_INCLUDE_PATH=${LibClang_INCLUDE_DIR}:/Library/Developer/CommandLineTools/usr/include/c++/v1/  CPLUS_INCLUDE_PATH=${LibClang_INCLUDE_DIR}:/Library/Developer/CommandLineTools/usr/include/c++/v1/
       ${CMAKE_BINARY_DIR}/source/binder --bind ""  -max-file-size=100000
-      --root-module ${stestnamenodot} --prefix ${CMAKE_CURRENT_BINARY_DIR}/ ${${stestname}_config_flag} --single-file  --annotate-includes  ${stestname}.hpp.include  
+      --root-module ${stestnamenodot} --prefix ${CMAKE_CURRENT_BINARY_DIR}/ ${${stestname}_config_flag} --single-file  --annotate-includes  ${CMAKE_CURRENT_BINARY_DIR}/${stestname}.hpp.include  
       -- -x c++ -std=c++11   -I . -I ${CMAKE_CURRENT_SOURCE_DIR}  -isystem /  -I ${CLANG_INCLUDE_DIRS}   -iwithsysroot${LibClang_INCLUDE_DIR}
       WORKING_DIRECTORY "${CMAKE_CURRENT_SOURCE_DIR}" VERBATIM)
   else()
     ADD_CUSTOM_TARGET(target${stestnamenodot}cpp
     #The first two expressions below are for older clang
     COMMAND C_INCLUDE_PATH=${LibClang_INCLUDE_DIR}   CPLUS_INCLUDE_PATH=${LibClang_INCLUDE_DIR}  ${CMAKE_BINARY_DIR}/source/binder --bind ""  -max-file-size=100000
-      --root-module ${stestnamenodot} --prefix ${CMAKE_CURRENT_BINARY_DIR}/ ${${stestname}_config_flag} --single-file  --annotate-includes  ${stestname}.hpp.include  
+      --root-module ${stestnamenodot} --prefix ${CMAKE_CURRENT_BINARY_DIR}/ ${${stestname}_config_flag} --single-file  --annotate-includes  ${CMAKE_CURRENT_BINARY_DIR}/${stestname}.hpp.include  
       -- -x c++ -std=c++11   -I . -I ${CMAKE_CURRENT_SOURCE_DIR}  -isystem /  -I ${CLANG_INCLUDE_DIRS}   -iwithsysroot${LibClang_INCLUDE_DIR}
       WORKING_DIRECTORY "${CMAKE_CURRENT_SOURCE_DIR}" VERBATIM)
   endif()

